store/src/db/mod.rs

```diff
@@ -49,24 +49,32 @@ impl Db {
             create_dir_all(p)?;
         }
 
-        let pool =
-            SqliteConfig::new(config.database_filepath.clone()).create_pool(Runtime::Tokio1)?;
-
-        let conn = pool.get().await?;
+        let pool = SqliteConfig::new(config.database_filepath.clone())
+            .builder(Runtime::Tokio1)?
+            .post_create(Hook::async_fn(move |conn, _| {
+                Box::pin(async move {
+                    // Feature used to support `IN` and `NOT IN` queries. We need to load this module
+                    // for every connection we create to the DB to support the queries we want to run
+                    let _ = conn
+                        .interact(|conn| array::load_module(conn))
+                        .await
+                        .map_err(|_| HookError::StaticMessage("Loading carray module failed"))?;
+
+                    Ok(())
+                })
+            }))
+            .build()?;
 
         info!(
             sqlite = format!("{}", config.database_filepath.display()),
             COMPONENT, "Connected to the DB"
         );
 
-        // Feature used to support `IN` and `NOT IN` queries
-        conn.interact(|conn| array::load_module(conn))
-            .await
-            .map_err(|_| anyhow!("Loading carray module failed"))??;
+        let conn = pool.get().await?;
 
         conn.interact(|conn| migrations::MIGRATIONS.to_latest(conn))
             .await
-            .map_err(|_| anyhow!("Migration task failed with a panic"))??;
+            .map_err(|_| HookError::StaticMessage("Migration task failed with a panic"))??;
```

Comment:
I would probably change this back to `anyhow` error - unless there is a good reason to use `HookError` here?